update_version.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
```

Comment:
This should probably live in a folder off of the root?  Maybe `utils/update_version.py`?

Reply:
Also, I believe we are going to need this script in all of our python projects we package for PyPi.

Reply:
> This is a fine PR. One question, can we get what `update_version.py` from another existing github action? This solution is fine, but I feel like we aren't the only project with this need.

You want `update_version.py` even when running locally. It could be a submodule (or maybe better a subtree?).

Reply:
I'm not sure the right solution.  `git subtree` might be good because then each repo has it with the submodule, but it can be updated?

At a minimum, lets move all of the version stuff into a sub-directory, and then if we make it a subtree, and can attach at that point.